mail_activity_plan/models/mail_activity_mixin.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2024 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from lxml import etree
+
+from odoo import _, api, fields, models
+
+from odoo.addons.base.models import ir_ui_view
+
+
+class MailActivityMixin(models.AbstractModel):
+    _inherit = "mail.activity.mixin"
+
+    has_activity_plans = fields.Boolean(
+        compute="_compute_has_activity_plans",
+    )
+
+    def _compute_has_activity_plans(self):
+        plan_model = self.env["mail.activity.plan"]
+        for item in self:
+            plans = plan_model._get_plans_from_model(item._name)
+            item.has_activity_plans = True if len(plans) else False
```

Comment:
Avoid spaghetti code:

```suggestion
            item.has_activity_plans = bool(plans)
```